### PR TITLE
Users

### PR DESCRIPTION
**Pull request by whwebdev**

### CODE REVIEW

**Comment by SpraggleRock** [comment]
@mattbaker @mikelikesbikes Greetings. Note: our feature test coverage is a bit spotty, we didn't write tests for things where we weren't sure of the final behavior we wanted.

File: TwitterStories/app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.create(user_params)
+    if user.save
```
Comment (mattbaker) [inline TwitterStories/app/controllers/users_controller.rb:9]: No need for save if you're using `#create`. You can use `#valid?` on your user object after `#create` to detect errors. Calling `#valid?` will populate the list of errors available via `#errors`.

File: TwitterStories/spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    it "redirects if user is created" do
+      User.should_receive(:create).with(params).and_return(our_user)
+
+      post(:create, user: params)
+      expect(response).to redirect_to user_path(our_user)
+    end
+
+    it "renders new user form if user isn't created" do
+      User.should_receive(:create).with(params).and_return(our_user)
+      our_user.should_receive(:save).and_return(false)
```
Comment (mattbaker) [inline TwitterStories/spec/controllers/users_controller_spec.rb:26]: See comment above about create/save

File: TwitterStories/spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+require 'spec_helper'
+
+describe UsersController do
```
Comment (mattbaker) [inline TwitterStories/spec/controllers/users_controller_spec.rb:3]: It'd be nice to have a test for your `#update` action

File: TwitterStories/spec/integration/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    fill_in('Username', with: 'superpoops')
+    fill_in('Password', with: 'shadowthemepoop')
+    fill_in('Password confirmation', with: 'shadowthemepoop')
+
+    click_button('Sign Up')
+
+    expect(current_path).to eq user_path(User.last)
+
+    expect(page).to have_content('WELCOME SUPERPOOPS!')
```
Comment (mattbaker) [inline TwitterStories/spec/integration/user_spec.rb:15]: I believe you can also assert that the `User` account increased by one in your feature spec.

**Pull request merged**